Dockerfile

```
@@ -84,9 +84,9 @@ RUN ldconfig
 
 # remove (dated) security workaround preventing use of
 # ImageMagick's convert on PDF/PS/EPS/XPS:
-RUN sed -i 's/rights="none"/rights="read|write"/g' /etc/ImageMagick-6/policy.xml
+RUN sed -i 's/rights="none"/rights="read|write"/g' /etc/ImageMagick-6/policy.xml || true
```

```suggestion
RUN test -f /etc/ImageMagick-6/policy.xml && sed -i 's/rights="none"/rights="read|write"/g' /etc/ImageMagick-6/policy.xml
```

---

Otherwise unexpected errors from `sed` would not be noticed.

---

> Otherwise unexpected errors from `sed` would not be noticed.

Which errors do you expect? The _only_ thing that can go wrong here is if the file is not present, or not readable/writable – all of which is covered by the original change. It's also more concise, showing that this step should never break the build.

---

The only acceptable condition is that the file is missing. Then there is no need to patch it of course.

If there is a problem with access permissions, that's a problem which should abort the process, because the code here fails to patch the file although that is required.

If it does not matter whether the existing file is patched or not, I suggest to remove the whole line.

---

> 
> 
> > Otherwise unexpected errors from `sed` would not be noticed.
> 
> Which errors do you expect? The _only_ thing that can go wrong here is if the file is not present, or not readable/writable – all of which is covered by the original change. It's also more concise, showing that this step should never break the build.

Why would the file not be present in the Docker image? We know exactly which base image we build upon, so we can be sure whether the image magick config files exist. If they do not, because the image changed e.g., then we should probably take notice by the build failing. So unless I overlook something, I'm against hiding such errors.

---

> Why would the file not be present in the Docker image? We know exactly which base image we build upon, so we can be sure whether the image magick config files exist. If they do not, because the image changed e.g., then we should probably take notice by the build failing. So unless I overlook something, I'm against hiding such errors.

Because not all module selections depend on IM. For example, if you exclude `ocrd_olena`, Docker will probably not have ImageMagick. (That Dockerfile is not only used for pre-built configs!)

---

Oh right, hadn't thought of that. Then `|| true` in those statements makes sense.

---

> The only acceptable condition is that the file is missing. Then there is no need to patch it of course.

This is the case (see above)

> If there is a problem with access permissions, that's a problem which should abort the process, because the code here fails to patch the file although that is required.

Agreed, but in our Docker build we are always priviledged, so this is actually out.

> If it does not matter whether the existing file is patched or not, I suggest to remove the whole line.

It does – IM (even the library) does not work otherwise, due to the overly restrictive defaults. (And I think IM's policy workaround to a security problem is a bad design choice...)